File: Visitor.java
class Visitor {

    //private Zoo zoo = new Zoo();
    private String name, lastName, birthdate, curp, admissionDate;
    private int numberVisits = 0, id, descuento; //NO PODEMOS INGRESAR VISITAS AHORA
    
    public Visitor (String name, String lastName, String birthdate, String curp, String admissionDate, int id) {
        this.name = name;
        this.lastName = lastName;
        this.birthdate = birthdate;
        this.curp = curp;
        this.admissionDate = admissionDate;
        this.id = id;
    }

    int getId() {
        return id;
    }

    String getName() {
        return name;
    }

    String getLastName() {
        return lastName;
    }

    void addNumberVisits() {
        this.numberVisits++;
        this.descuento++;
        if (descuento == 6) {
            this.descuento = 1;
        }
    }

    int getNumberVisits() {
        return descuento;
    }

    int getAgeBirthdate() { //obtiene el año de la fecha de nacimiento
        String[] parts = birthdate.split("-");
        return Integer.parseInt(parts[2]);
    }

    int getBirthdate(){
        String date;
        String[] parts = birthdate.split("-");
        date = parts[1]+parts[0];
        return Integer.parseInt(date);
    }

    //CAMBIOS EN EL GET DATA
    String visitorGetData() { //datos de visitante
        return String.format("\nNombre: %s %s \nID: %d \nFecha de nacimiento: %s \nCURP: %s \nNúmero de visitas: %d \nFecha de ingreso: %s", 
            this.name, this.lastName, this.id, this.birthdate, this.curp, this.numberVisits, this.admissionDate);
    }

}

File: Zoo python/Visitor.py
class Visitor:
    def __init__(self, name, lastName, birthdate, curp, admissionDate, id):
        self.name = name
        self.lastName = lastName
        self.birthdate = birthdate
        self.curp = curp
        self.admissionDate = admissionDate
        self.id = id
        self.numberVisits = 0
        self.descuento = 0

    def getId(self):
        return self.id

    def getName(self):
        return self.name

    def getLastName(self):
        return self.lastName

    def addNumberVisits(self):
        self.numberVisits += 1
        self.descuento += 1
        if self.descuento == 6:
            self.descuento = 1

    def getNumberVisits(self):
        return self.descuento

    def getAgeBirthdate(self):
        parts = self.birthdate.split("-")
        return int(parts[2])

    def getBirthdate(self):
        parts = self.birthdate.split("-")
        date = parts[1] + parts[0]
        return int(date)

    def setId(self, id):
        self.id = id

    def setName(self, name):
        self.name = name

    def setLastName(self, lastName):
        self.lastName = lastName

    def setBirthdate(self, birthdate):
        self.birthdate = birthdate

    def setAdmissionDate(self, admissionDate):
        self.admissionDate = admissionDate

    def setCurp(self, curp):
        self.curp = curp

    def visitorGetData(self):
        return f"\nNombre: {self.name} {self.lastName} \nID: {self.id} \nFecha de nacimiento: {self.birthdate} \nCURP: {self.curp} \nNúmero de visitas: {self.numberVisits} \nFecha de ingreso: {self.admissionDate}"
